utils/utils_epd.py

```diff
@@ -0,0 +1,146 @@
+# This file is released under MIT licence, see file LICENSE.
+# Author(s):       Theo Lacombe
+#
+# Copyright (C) 2021 Inria
+
+import numpy as np
+import matplotlib.pyplot as plt
+from matplotlib.patches import Polygon
+import gudhi as gd
+
+def sample_circle(n):
+    theta = 2 * np.pi * np.random.rand(n)
+    x, y = np.cos(theta), np.sin(theta)
+    return np.array([x, y]).T
+
+
+def sample_torus(n, r1, r2):
+    theta1 = 2 * np.pi * np.random.rand(n)
+    theta2 = 2 * np.pi * np.random.rand(n)
+
+    x = (r1 + r2 * np.cos(theta2)) * np.cos(theta1)
+    y = (r1 + r2 * np.cos(theta2)) * np.sin(theta1)
+    z = r2 * np.sin(theta2)
+
+    X = np.array([x, y, z]).T
+
+    return X
+
+
+def sample_noise(N, ndim, scale=1., type="uniform"):
+    '''
+    noise sample uniforme on [-scale, scale]^2 or gaussien N(0, scale^2)
+    '''
+    if type=="uniform":
+        X_noise = scale * (2 * np.random.rand(N, ndim) - 1)
+    elif type=="gaussian":
+        X_noise = scale * np.random.randn(N, ndim)
+    return X_noise
+
+
+def add_noise(X_noise, X):
+    return np.concatenate([X, X_noise])
+
+
+# Compute persistence diagram using the Cech filtration
+def alphacomplex(X, hdim=1):
+    ac = gd.AlphaComplex(points=X)
+    st = ac.create_simplex_tree()
+    pers = st.persistence(min_persistence=0.0001)
+    h1 = st.persistence_intervals_in_dimension(hdim)
+    return np.sqrt(np.array(h1))  # we must sqrt to get Cech dgm
+
+
+def expected_dgm(X, n, k, hdim=1, replace=True):
```

Comment:
name doesn't match what it does?

Reply:
You mean renaming the function `cech_dgm` for instance ? Indeed that may be better.

Reply:
It returns a list of diagrams, built from subsamples. I assume that originally it was also computing some average, but was later changed to return the list. cech_dgm would also be confusing. Anyway that's just a detail.